lib/iaas/AzureClient.js

```diff
@@ -0,0 +1,192 @@
+          .chain(file)       
+          .pick('name', 'lastModified')       
+          .set('isTruncated', isTruncated)       
+          .value()          
+        ));      
+        return files;    
+      });
+  }
+
+  remove(container, file) {
+    if (arguments.length < 2) {
+      file = container;
+      container = this.containerName;
+    }
+    logger.debug(`Deleting file ${file} in container ${container} `);
+    return this.storage
+      .deleteBlobAsync(container, file)
+      .then(response => logger.info(`Response of blob deletion from cloud storage for ${file} :`, response))
+      .catch(BaseCloudClient.providerErrorTypes.Unauthorized, err => {
+        logger.error(err.message);
+        throw new Unauthorized('Authorization at the cloud storage provider failed');
+      })
+      .catch(BaseCloudClient.providerErrorTypes.Forbidden, err => {
+        logger.error(err.message);
+        throw new Forbidden('Authentication at the cloud storage provider failed');
+      })
+      .catchThrow(BaseCloudClient.providerErrorTypes.NotFound, new NotFound(`Object '${file}' not found`));
+  }
+
+  download(options) {
+    return utils.streamToPromise(this.storage.createReadStream(
+        options.container, options.remote))
+      .catchThrow(BaseCloudClient.providerErrorTypes.NotFound, new NotFound(`Object '${options.remote}' not found`));
+  }
+
+  upload(options, buffer) {
+    return new Promise((resolve, reject) => {
+      function cleanup() {
+        uploadedStream.removeListener('finish', onfinish);
+        uploadedStream.removeListener('error', onerror);
+      }
+
+      function onerror(err) {
+        cleanup();
+        reject(err);
+      }
+
+      function onfinish() {
+        cleanup();
+        resolve(JSON.parse(buffer));
+      }
+
+      let uploadedStream = this.storage
+        .createWriteStreamToBlockBlob(
+          options.container, options.remote);
+
+      uploadedStream.on('error', onerror);
+      uploadedStream.on('finish', onfinish);
+      uploadedStream.end(buffer);
+    });
+  }
+
+  uploadJson(container, file, data) {
+    if (arguments.length < 3) {
+      data = file;
+      file = container;
+      container = this.containerName;
+    }
+    return this
+      .upload({
+        container: this.containerName,
+        remote: file
+      }, new Buffer(JSON.stringify(data, null, 2), 'utf8'));
+  }
+
+  downloadJson(container, file) {
+    if (arguments.length < 2) {
+      file = container;
+      container = this.containerName;
+    }
+    return this
+      .download({
+        container: this.containerName,
+        remote: file
+      })
+      .then((data) => {
+        const response = JSON.parse(data);
+        response.trigger = response.trigger === CONST.BACKUP.TRIGGER.MANUAL ? CONST.BACKUP.TRIGGER.SCHEDULED : response.trigger;
+        //The above conversion is done to handle existing CRON Jobs which set this trigger as 'manual' even for scheduled Jobs
```

Comment:
Remove these comments as its not applicable in case of AZURE and also these checks are no longer required. This was only required in case of earier implementation where you had scheduled backups being set as manual in meta info.

Reply:
done